src/twisted/test/test_twistd.py

```diff
@@ -994,7 +994,9 @@ def setgid(gid):
         self.patch(os, "setgid", setgid)
 
         options = twistd.ServerOptions()
-        options.parseOptions(["--nodaemon", "--uid", str(wantedUid)])
+        options.parseOptions(
+            ["--nodaemon", "--uid", str(wantedUid), "--pidfile={}".format(pidFile)]
+        )
         application = service.Application("test_setupEnvironment")
         self.runner = UnixApplicationRunner(options)
         runner = UnixApplicationRunner(options)
```

Comment:
I think that one problem is that the tests are starting the application here, but they don't stop the application at the end of the test.

Not sure what API should be used to stop an application